docs/platform_developers/create_server_extension.rst

```diff
@@ -0,0 +1,127 @@
+*******************************
+Creating a new server extension
+*******************************
+
+An Inmanta server extension adds extra functionality to the Inmanta core. Each extension consists of one or more server slices
+that provide certain functionality. This page describes how to create a new extension and the associated server slice(s).
+
+
+The package layout of a server extension
+########################################
+
+Each Inmanta server extension is defined as a subpackage of the ``inmanta_ext`` package. The following directory structure is
```

Comment:
I would add something like this after the first sentence: ``inmanta_ext`` is a namespace package used by the service orchestrator to discover new extensions.